components/figure/Renderer.tsx

```diff
@@ -50,12 +50,20 @@ const Renderer: React.FC<RendererProps> = ({ document, rawDocument }) => {
     }
   }, []);
 
-  const handlePrint = useCallback(() => {
-    if (
-      navigator.userAgent.includes("SamsungBrowser") ||
-      navigator.userAgent.includes("CriOS") ||
-      navigator.userAgent.includes("FxiOS")
-    ) {
+  const handlePrint = useCallback(async () => {
+    const ua = window.navigator.userAgent;
+    const isIOS = /iPad|iPhone|iPod/.test(ua);
+    // List of common browsers installable from app store
+    const isNotSafari = /CriOS|FxiOS|EdgiOS|YaBrowser|OPT|OPR/.test(ua);
+    const isSamsungBrowser = ua.includes("SamsungBrowser");
+    // https://stackoverflow.com/questions/36523448/how-do-i-tell-if-a-user-is-using-brave-as-their-browser/60954062#60954062
+    // no typings for browser-specific fields - https://github.com/microsoft/TypeScript/issues/41532
+    const isBraveBrowser =
+      ((window.navigator as any).brave && (await (window.navigator as any).brave.isBrave())) || false;
+
+    const isUnsupportedBrowser = (isIOS && (isNotSafari || isBraveBrowser)) || isSamsungBrowser;
```

Comment:
Since only the Safari browser is supported on iOS devices, why not we just whitelist `isIOS` && `!isSafari` and reject all others?

In that way, we won't have to detect all possible browsers a user may use in an iOS device.

Something like this:

```typescript
const isIOS = /iPad|iPhone|iPod/i.test(ua);
const isSafari = /Safari|AppleWebKit/i.test(ua);

const isUnsupportedBrowser = (isIOS && !isSafari) || isSamsungBrowser;
```

Reply:
Oh yeah, I tried to find a unique identifier for Safari but there wasn't a way to detect specifically iOS Safari. 

E.g. for Chrome on iPhone user agent, `Mozilla/5.0 (iPhone; CPU iPhone OS 16_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/111.0.5563.72 Mobile/15E148 Safari/604.1`, it has keywords like "Safari", "AppleWebKit", which Safari user agent also has.

So we can only have a list of non-safari browsers to filter by.

Reply:
Ok I see! Can stick to check non-Safari browsers then